sandbox/modules/identity/src/test/java/org/opensearch/identity/BearerAuthTests.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.shiro.authc.AuthenticationToken;
+import org.hamcrest.MatcherAssert;
+import org.opensearch.authn.jwt.JwtVendor;
+import org.opensearch.authn.jwt.JwtVerifier;
+import org.opensearch.authn.tokens.BearerAuthToken;
+import org.opensearch.client.Request;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.authn.AuthenticationTokenHandler.extractShiroAuthToken;
+
+public class BearerAuthTests extends AbstractIdentityTestCase {
+
+    public void testExpiredValidJwt() {
+
+        Map<String, String> jwtClaims = new HashMap<>();
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendor.createExpiredJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+        BearerAuthToken bearerAuthToken = new BearerAuthToken(headerBody);
+
+        try {
+            AuthenticationToken extractedShiroToken = extractShiroAuthToken(bearerAuthToken);
+        } catch (RuntimeException ex) {
+            assertFalse(ex.getMessage().isEmpty());
+            assertEquals("The token has expired", ex.getMessage());
+        }
+    }
+
+    public void testEarlyValidJwt() {
+
+        Map<String, String> jwtClaims = new HashMap<>();
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendor.createEarlyJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+        BearerAuthToken bearerAuthToken = new BearerAuthToken(headerBody);
+
+        try {
+            AuthenticationToken extractedShiroToken = extractShiroAuthToken(bearerAuthToken);
+        } catch (RuntimeException ex) {
+            assertFalse(ex.getMessage().isEmpty());
+            assertEquals("The token cannot be accepted yet", ex.getMessage());
+        }
+    }
+
+    public void testValidJwt() {
+
+        Map<String, String> jwtClaims = new HashMap<>();
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendor.createJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+        BearerAuthToken bearerAuthToken = new BearerAuthToken(headerBody);
+        AuthenticationToken extractedShiroToken;
+
+        try {
+            extractedShiroToken = extractShiroAuthToken(bearerAuthToken); // This should verify and then extract the shiro token for login
+            assertEquals(encodedToken, extractedShiroToken.getPrincipal());
+        } catch (RuntimeException ex) {
+            throw new Error(ex);
+        }
+        if (extractedShiroToken == null) {
+            throw new Error("The value of the extracted token is null.");
+        }
+    }
+
+    public void testInvalidJwt() {
```

Comment:
Nice test! wdyt about making the name more descript to indicate that the JWT was signed with a unexpected (mismatched) algorithm?

Reply:
Yeah that sounds like a good idea